lib/src/lifecycle_module.dart

```diff
@@ -596,14 +618,19 @@ abstract class LifecycleModule extends SimpleModule with Disposable {
           ]);
     }
 
-    var pendingTransition = _transition?.future;
+    Future pendingTransition;
+    if (_transition != null && !_transition.isCompleted) {
+      pendingTransition = _transition.future;
+    }
+
     _previousState = _state;
     _state = LifecycleState.unloading;
-    _transition = new Completer<Null>();
+    final transition = new Completer<Null>();
+    _transition = transition;
 
     var unloadAndDispose = new Completer<Null>();
-    unloadAndDispose.complete(_transition.future.then((_) => dispose()));
-    _transition.complete(_unload(pendingTransition));
+    unloadAndDispose.complete(transition.future.then((_) => dispose()));
+    transition.complete(_unload(pendingTransition));
```

Comment:
This one was already setup in the way I suggested above, so again the extra `transition` var is not needed.